Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <SdfAssetBuilder/SdfAssetBuilder.h>
+
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/IOUtils.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+#include <AzToolsFramework/Entity/EntityUtilityComponent.h>
+#include <AzToolsFramework/Prefab/PrefabLoaderScriptingBus.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+#include <AzToolsFramework/Prefab/PrefabSystemScriptingBus.h>
+#include <AzToolsFramework/Prefab/Procedural/ProceduralPrefabAsset.h>
+
+#include <AssetBuilderSDK/AssetBuilderSDK.h>
+#include <AssetBuilderSDK/SerializationDependencies.h>
+
+namespace ROS2
+{
+        namespace
+        {
+            [[maybe_unused]] constexpr const char* SdfAssetBuilderName = "SdfAssetBuilder";
+            constexpr const char* SdfAssetBuilderJobKey = "Sdf Asset Builder";
+            constexpr const char* SdfAssetBuilderSupportedFileExtensionsRegistryKey = "/ROS/SdfAssetBuilder/SupportedFileTypeExtensions";
```

Comment:
Should we place this key under `/O3DE/ROS2` instead?

Any insight @adamdbrw on how ROS2 settings are catagorized?
```suggestion
            constexpr const char* SdfAssetBuilderSupportedFileExtensionsRegistryKey = "/O3DE/ROS2/SdfAssetBuilder/SupportedFileTypeExtensions";
```

Reply:
If ROS2 or Robotics both sound good, I'll go with the proposed /O3DE/ROS2 since that matches up with the Gem name.